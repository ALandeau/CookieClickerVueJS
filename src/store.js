import Vue from 'vue'
import VueCookie from 'vue-cookie'
import Vuex from 'vuex'
import { atoutStore } from "./atoutstore";

Vue.use(Vuex);

export default new Vuex.Store({
    state: {
        scoreNumber: VueCookie.get('score'),
        globalTimer: 0
    },
    getters: {
        scoreNumber: state => state.scoreNumber,
    },
    mutations: {
        incrementScoreNumber(state) {
            state.scoreNumber++;
            VueCookie.set('score', state.scoreNumber);
        },
        autoIncrementScoreNumber(state) {
            if(state.atoutStore.grandmaNumber) state.globalTimer = state.globalTimer + +state.atoutStore.grandmaNumber;
            if(state.atoutStore.cursorNumber) state.globalTimer = state.globalTimer + (+state.atoutStore.cursorNumber / 10);
            setInterval(function () {state.scoreNumber = +state.scoreNumber + +state.globalTimer}, 1000);
        },
        grandmaInteract(state) {
            let grandmaPrice = state.atoutStore.grandmaPrice;
            state.scoreNumber = state.scoreNumber - grandmaPrice;
            state.atoutStore.grandmaNumber++;
            grandmaPrice = +grandmaPrice * (((state.atoutStore.grandmaNumber / 10)) + 1);
            grandmaPrice = Math.round(grandmaPrice);
            state.atoutStore.grandmaPrice = grandmaPrice;
            VueCookie.set('grandma', state.atoutStore.grandmaNumber)
        },
        cursorInteract(state) {
            let cursorPrice = state.atoutStore.cursorPrice;
            state.scoreNumber = state.scoreNumber - cursorPrice;
            state.atoutStore.cursorNumber++;
            cursorPrice = +cursorPrice * (((state.atoutStore.cursorNumber / 10)) + 1);
            cursorPrice = Math.round(cursorPrice);
            state.atoutStore.cursorPrice = cursorPrice;
            VueCookie.set('cursor', state.atoutStore.cursorNumber)
        }
    },
    modules: {
        atoutStore,
    },
});