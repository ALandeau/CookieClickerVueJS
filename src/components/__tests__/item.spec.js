import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Item from "@/components/itemCookie.vue";
import store from "@/store";

describe("itemCookie.vue", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  const wrapper = mount(Item, {
    store,
    localVue
  });

  it("renders the correct markup", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("bluecursor click should correctly enable", () => {
    store.state.scoreNumber = 1000;
    expect(store.state.scoreNumber).toBe(1000);
    const counter = wrapper.find("#item-bluecursor div.enable");
    counter.trigger("click");
    wrapper.find("#atout-grandma div.disable");
    expect(store.state.scoreNumber).toBeLessThan(1000);
  });
});
