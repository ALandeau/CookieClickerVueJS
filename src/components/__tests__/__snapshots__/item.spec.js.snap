// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemCookie.vue renders the correct markup 1`] = `
<div id="item">
  <div class="show-number-atout"></div>
  <div id="item-bluecursor" class="item-element">
    <div class="add-element disable">
      <div class="element-image">
        <img src="../../assets/images/bluecursor_disable.png" alt="blue cursor disable">
      </div>
      <div class="element-wrapper">
        <div class="element-price">
          <p class="image-center">
            <img src="../../assets/images/perfectCookie.png">
          </p>
          <p class="price-disable">1000</p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
